assets/src/stories-editor/helpers/index.js

```diff
@@ -1868,33 +1870,23 @@ export const copyTextToClipBoard = ( text ) => {
  * Ensure that only allowed blocks are pasted.
  *
  * @param {[]}      blocks Array of blocks.
- * @param {string}  clientId Page ID.
- * @param {boolean} isFirstPage If is first page.
+ * @param {string}  pageId Page ID.
  * @return {[]} Filtered blocks.
  */
-export const ensureAllowedBlocksOnPaste = ( blocks, clientId, isFirstPage ) => {
-	const allowedBlocks = [];
-	blocks.forEach( ( block ) => {
-		switch ( block.name ) {
-			// Skip copying Page.
-			case 'amp/amp-story-page':
-				return;
-			case 'amp/amp-story-page-attachment':
-			case 'amp/amp-story-cta':
-				const currentBlock = getPageBlockByName( clientId, block.name );
-				if ( currentBlock || ( isFirstPage && block.name === 'amp/amp-story-cta' ) ) {
-					return;
-				}
-				allowedBlocks.push( block );
-				break;
-			default:
-				if ( ALLOWED_CHILD_BLOCKS.includes( block.name ) ) {
-					allowedBlocks.push( block );
-				}
-				break;
-		}
-	} );
-	return allowedBlocks;
+export const ensureAllowedBlocksOnPaste = ( blocks, pageId ) =>
+	blocks.filter( ( block ) => isBlockAllowedOnPage( block.name, pageId ) );
+
+/**
+ * Is the given block allowed on the given page?
+ *
+ * @param {Object}  name The name of the block to test.
+ * @param {string}  pageId Page ID.
+ * @return {boolean} Returns true if the element is allowed on the page, false otherwise.
+ */
+export const isBlockAllowedOnPage = ( name, pageId ) => {
```

Comment:
Most invocations of this function have `pageId` set to current page. Should it just default to this perhaps (if null, set to `getCurrentPage()`), reducing the complexity of those calls?

Reply:
I was considering it, but would then have to mock the `select`'ed functions making it a bit tricky, but of course possible.

I'll take a swing at it and see how cumbersome it is.

Reply:
It required reorganising things a bit, but it's a lot cleaner now: 3c7afcf.

Reply:
I did not actually default `pageId` to `getCurrentPage()`. It's of course still possible. Thoughts?

Reply:
I think it's fine to keep that as-is, there'd be only a little benefit.

Nice work on the tests!